### ISSUE TITLE
Reuse .NET core system.io.compression

### ISSUE THREAD

**Issue by jdehaan** [jdehaan/SafeRapidPdf#6]
I would like to replace the existing decompression code with something like this:

```cs
byte[] decompressed;
using (var zin = new DeflateStream(new MemoryStream(Data.Data), CompressionMode.Decompress))
{
        using (var outStream = new MemoryStream())
        {
            zin.CopyTo(outStream);
            decompressed = outStream.ToArray();
        }
}
```

In order to not pull in too much code and reuse .net core components. The problem is that this triggers an `InvalidDataException`. I could not find out what is wrong. I believe .NET core expects some header part around the raw data. (just an idea, unverified)

**Comment by iamcarbon** [comment]
@jdehaan Do you happen to have a PDF that hits this path that we can add as a test case? I'll take a stab at this.

**Comment by jdehaan** [comment]
I'll provide you the file, I do not have my set of test files here unfortunately. We should then also add that into the repos for the unit tests (I then strip the pdf down to the interesting part, which is a bit of effort but then we are sure the unit test tests only that part).

**Comment by jdehaan** [comment]
I had a look again inside the tests, `Decode_FlateDecode_PngPredictor_Up` already tests the basic deflate code based on an example. Your new code is straight and clean. I do not know how I could have had issues by looking at the result... :-)

I used a very located part of a pdf in order to test only that particular code, instead of example pdfs which would be more system testing than unit testing, due to the fact they cover a wide range of additional features tested along with the actual thing we want to test. I checked your code was hit during the execution of the test by setting a breakpoint.

Thanks for your contributions!
